### PR TITLE
Scatter plots

### PR DESCRIPTION
**Pull request by karukes**
- I created the correlation plot: a number of training data points vs metric.
- Methods/Plots  of the box plots and heatmap plots are edited to solve issue #155

### CODE REVIEW

File: main_plot.py
Comment (swhustla) [inline main_plot.py:1]: @karukes does this file need a bit of tidying up? It looks like you have disabled a bunch of plot types. Or am I mistaken?

File: methods/plot_correlation_Npoints.py
Comment (swhustla) [inline methods/plot_correlation_Npoints.py:1]: File naming should conform to PEP8 standards (check with chatGPT)

File: methods/plot_results_in_box_plot.py [hunk 1]
```diff
@@ -21,44 +21,50 @@
...
             figure_method = plot_boxplot_by_method(data_to_plot, chosen_metric)
             figure_city = plot_boxplot_by_city(data_to_plot, chosen_metric)
-            stock_company_name = data_to_plot["subset_row"][0]
+            plot_params = get_plot_params(
+                figure_method, chosen_metric, data_to_plot, dataset_name
+            )
+            plot_params = get_plot_params(
+                figure_city, chosen_metric, data_to_plot, dataset_name
+            )
             print(f"Saving {dataset_name} {chosen_metric} boxplot plot")
```
Comment (swhustla) [inline methods/plot_results_in_box_plot.py:38]: Isn't there a duplication of writing to plot_params here?

File: methods/plot_results_in_box_plot.py [hunk 2]
```diff
@@ -21,44 +21,50 @@
...
+            #         "by_data",
+            #     ),
+            # }
+            # input_key = (dataset_name, stock_company_name)
+            # if input_key in input_map:
+            #     figure_params = input_map[input_key]
+            save_plot_boxplot(*plot_params[:-1]),
+            save_plot_boxplot(
+                figure_city, plot_params[1], plot_params[4], chosen_metric
+            ),
```
Comment (swhustla) [inline methods/plot_results_in_box_plot.py:68]: Can you add comments to explain what is going on here? Why plotting twice to boxplot?

File: plots/plot_correlation_Npoints.py
Comment (swhustla) [inline plots/plot_correlation_Npoints.py:1]: File naming needs to be PEP8 standard - see earlier comment

File: plots/plot_correlation_Npoints.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    # deal with the case where there are no results for a method
+    pivoted_dataframe = pivoted_dataframe.fillna(0)
+    data_lenght = []
+    for data in training_data:
+        data_lenght.append(len(data))
+    pivoted_dataframe_Ndata = pivoted_dataframe.copy()
+    pivoted_dataframe_Ndata["Ndata"] = data_lenght
+
+    if chosen_metric == "R2":
+        # create more ticks on the x axis
```
Comment (swhustla) [inline plots/plot_correlation_Npoints.py:68]: Maybe get ChatGPT to help you restructure some of this, it seems slightly repetitive code.

**Review by swhustla — CHANGES_REQUESTED** [review]
Hi @karukes a few comments and feedback for you :)

File: main_plot.py
Comment (karukes) [reply main_plot.py:1]: @swhustla it is disabled in order not to make all the plots at once (without need). Agree, maybe this part could be organised bette.

**Review by swhustla — APPROVED** [review]
Looks good, thanks for the quick changes

**Pull request merged**